packages/webhook-utils/src/webhook-updater.ts

```diff
@@ -0,0 +1,158 @@
+import { WebhookManifest } from "@saleor/app-sdk/types";
+import { Client } from "urql";
+import { createAppWebhookFromManifest } from "./create-app-webhook-from-manifest";
+import { createAppWebhookFromWebhookDetailsFragment } from "./create-app-webhook-from-webhook-details-fragment";
+import { getWebhookIdsAndManifestsToUpdate } from "./filters/get-webhook-ids-and-manifests-to-update";
+import { webhooksToAdd } from "./filters/webhooks-to-add";
+import { webhooksToRemove } from "./filters/webhooks-to-remove";
+import { modifyAppWebhookFromManifest } from "./modify-app-webhook-from-manifest";
+import { modifyAppWebhookFromWebhookDetails } from "./modify-app-webhook-from-webhook-details";
+import { removeAppWebhook } from "./operations/remove-app-webhook";
+import { Logger, WebhookData } from "./types";
+
+export class WebhookUpdater {
+  private dryRun: boolean;
+  private logger: Logger;
+  private client: Client;
+  private webhookManifests: Array<WebhookManifest>;
+  private existingWebhooksData: Array<WebhookData>;
+  private addedWebhooks: Array<WebhookData>;
+  private modifiedWebhooks: Array<WebhookData>;
+  private removedWebhooks: Array<WebhookData>;
+
+  constructor(args: {
+    dryRun: boolean;
+    logger: Logger;
+    client: Client;
+    webhookManifests: Array<WebhookManifest>;
+    existingWebhooksData: Array<WebhookData>;
+  }) {
+    this.dryRun = args.dryRun;
+    this.logger = args.logger;
+    this.client = args.client;
+    this.webhookManifests = args.webhookManifests;
+    this.existingWebhooksData = args.existingWebhooksData;
+    this.addedWebhooks = [];
+    this.modifiedWebhooks = [];
+    this.removedWebhooks = [];
+  }
+
+  public update = async () => {
+    this.logger.debug("Updating webhooks");
+
+    try {
+      this.logger.debug("Preparing list of changes to be executed");
+
+      // Based on names, find the ones which should be added
+      const webhookManifestsToBeAdded = webhooksToAdd({
```

Comment:
An idea - here we patch webhooks, which looks like error prone (or not)?

Alternative solution to think about

1. Add every webhook from current manifest 
2. Disable old webhooks
3. Enable new webhooks
4. Remove old ones

Not sure which will be safer, best if we have transactional batch update available